File: src/main/java/io/github/oleksivio/telegram/bot/api/model/objects/passport/errors/PassportElementErrorFiles.java
package io.github.oleksivio.telegram.bot.api.model.objects.passport.errors;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.oleksivio.telegram.bot.api.model.objects.passport.PassportConstants;

import java.util.ArrayList;
import java.util.List;

/**
 * @see <a href="https://core.telegram.org/bots/api#passportelementerrorfiles">PassportElementErrorFiles</a>
 */
public class PassportElementErrorFiles extends PassportElementError {
    private static final String FILES = "files";

    /**
     * file_hashes Array of String List of base64-encoded file hashes
     */
    @JsonProperty("file_hashes")
    private List<String> fileHashes = new ArrayList<>();

    public PassportElementErrorFiles() {
        super(FILES);
    }

    public List<String> getFileHashes() {
        return fileHashes;
    }

    public void setFileHashes(List<String> fileHashes) {
        this.fileHashes = fileHashes;
    }

    public void setType(Type type) {
        setType(type.stringName());
    }

    public enum Type {
        UTILITY_BILL(PassportConstants.UTILITY_BILL),
        BANK_STATEMENT(PassportConstants.BANK_STATEMENT),
        RENTAL_AGREENENT(PassportConstants.RENTAL_AGREENENT),
        PASSPORT_REGISTRATION(PassportConstants.PASSPORT_REGISTRATION),
        TEMPORARY_REGISTRATION(PassportConstants.TEMPORARY_REGISTRATION);

        private final String name;

        Type(String name) {
            this.name = name;
        }

        public String stringName() {
            return name;
        }
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/objects/passport/errors/PassportElementErrorFiles.kt
package io.github.oleksivio.tl.kbot.server.api.objects.passport.errors

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.objects.passport.PassportConstants
import io.github.oleksivio.tl.kbot.server.api.model.type.NamedType
import java.util.*

/**
 *  [PassportElementErrorFiles](https://core.telegram.org/bots/api/#passportelementerrorfiles)
 */
class PassportElementErrorFiles : PassportElementError() {
    @JsonProperty("source")
    val source: String = "files"
    /**
     * file_hashes Array of String List of base64-encoded file hashes
     */
    @JsonProperty("file_hashes")
    var fileHashes: List<String> = ArrayList()

    fun setType(type: Type) {
        this.type = type.typeName
    }

    enum class Type constructor(override val typeName: String) :
        NamedType {
        UTILITY_BILL(PassportConstants.UTILITY_BILL),
        BANK_STATEMENT(PassportConstants.BANK_STATEMENT),
        RENTAL_AGREENENT(PassportConstants.RENTAL_AGREENENT),
        PASSPORT_REGISTRATION(PassportConstants.PASSPORT_REGISTRATION),
        TEMPORARY_REGISTRATION(PassportConstants.TEMPORARY_REGISTRATION);

        override fun toString(): String {
            return typeName
        }

    }

}
